s3same/iam.py

```diff
@@ -0,0 +1,88 @@
+import json
+from botocore.exceptions import ClientError
+
+IAMName = 's3same_travis'
+
+def _policy_string(bucket):
+    return json.dumps({
+        "Version": "2012-10-17",
+        "Statement": [
+            {
+                "Action": [
+                    "s3:ListBucket"
+                    ],
+                "Effect": "Allow",
+                "Resource": [
+                    "arn:aws:s3:::{}".format(bucket)
+                    ]
+                },
+            {
+                "Action": [
+                    "s3:PutObject",
+                    "s3:PutObjectAcl"
+                    ],
+                "Effect": "Allow",
+                "Resource": [
+                    "arn:aws:s3:::{}/*".format(bucket)
+                    ]
+                }
+            ]
+        })
+
+def _all_policies(iam, **kwargs):
+    marker = None
+    while True:
+        if marker:
+            kwargs['Marker'] = marker
+        response = iam.list_policies(**kwargs)
+        for policy in response.get('Policies', []):
+            yield policy
```

Comment:
Generators inside generators. I'd just roll the if condition from below into this function, so:

```python
for policy in response.get('Policies', []):
    if policy.get('PolicyName') == IAMName:
        yield policy.get['Arn']
```